### ISSUE TITLE
[BUG] Accessibility Identifiers missing in v5.4

### ISSUE THREAD

**Issue by Natelegreat1** [Adyen/adyen-ios#1444]
**Describe the bug**
Our UI tests are failing after updating to v5.4 because the accessibility identifiers in the DropIn component are no longer found. Reverting back to v5.3 has the identifiers available.

**To Reproduce**
1. Run a UI test which opens the DropIn component
2. Try to access the _staticText_ `AdyenDropIn.PaymentMethodListComponent.Credit Card.titleLabel` when the DropIn is visible*
3. Note time out / XCUITest unable to find the label

*Note: I checked and there is also no _Button_ with that accessibility identifier either.

![Screenshot 2023-11-07 at 11 44 40](https://github.com/Adyen/adyen-ios/assets/2666640/a75010d1-98f3-415a-b8aa-66965520495a)

**Expected behavior**
`XCTAssertTrue(staticTexts[AdyenDropIn.PaymentMethodListComponent.Credit Card.titleLabel].firstMatch.exists)` should succeed.

**Environment**
 - Device: Any iOS device (iPhone, iPad)
 - iOS Version: iOS 15+
 - SDK Version: 5.4.0
 - Package management system: Swift Package Manager
 - Xcode version: 15.0.1
 - Mac OS type: M1 Pro running Sonoma 14.0

**Relevant source code**
This is the failure output as it tries to find the Credit Card label when the DropIn is visible.
```
Failed to Error Domain=com.apple.dt.xctest.ui-testing.error Code=10008 "No matches found for Elements matching predicate '"AdyenDropIn.PaymentMethodListComponent.Credit Card.titleLabel" IN identifiers' from input {(
    StaticText, label: 'Checkout',
    StaticText, label: 'Payment Methods', <--- here it is!
    StaticText, label: 'Subtotal, $5.00', value: $5.00,
    StaticText, label: 'Fees, $0.00', value: $0.00,
    StaticText, identifier: '*.Checkout.taxPriceLabel', label: 'Taxes, $0.00', value: $0.00,
    StaticText, identifier: '*.Checkout.totalPriceLabel', label: 'Total, $5.00', value: $5.00,
    StaticText, label: 'Details',
    StaticText, label: 'Subtotal',
    StaticText, label: '$5.00', value: $5.00,
    StaticText, label: 'Fees',
    StaticText, label: '$0.00', value: $0.00,
    StaticText, label: 'Taxes',
    StaticText, identifier: '*.Checkout.taxPriceLabel', label: '$0.00', value: $0.00,
    StaticText, label: 'Total',
    StaticText, identifier: '*.Checkout.totalPriceLabel', label: '$5.00', value: $5.00,
    StaticText, label: 'John Doe',
    StaticText, label: 'Continue to payment'
)}
```

**Comment by goergisn** [comment]
Hi @Natelegreat1

Thanks for getting in touch and sorry for the inconvenience this caused.

Generally it's not ideal to rely on accessibility identifiers as they are based on / expose internal implementation details and thus are not meant to be used outside of the SDK as they might change with any release even though it might not introduce API changes.

What we recommend is accessing the elements by static text instead. 
You can get a list of all accessible elements by using `debugDescription` of `XCUIElement` or `XCUIElementQuery`.

e.g. 
```swift 
print(app.debugDescription)
// or
print(app.staticTexts.debugDescription)
```

In your case that'd translate to
```swift
XCTAssertTrue(app.staticTexts["Credit Card"].firstMatch.exists)
// or
XCTAssertTrue(app.otherElements["Credit Card"].firstMatch.exists)
```

Best,
Alex

**Comment by Natelegreat1** [comment]
Shucks! Well that's a bummer. Using accessibility identifiers is much more reliable in case we are testing different locales or if the copy changes (etc. "Credit card" --> "Credit Cards") and it shouldn't affect UI tests.

We'll hope that the text is static and proceed as you described.

Thanks for confirming it was intentional and not a mistake!
